addons/pos_self_order/static/src/app/models/line.js

```diff
@@ -43,7 +43,19 @@ export class Line extends Reactive {
 
         this.initCustomAttribute();
     }
+    extractProductNameAndAttributes() {
+        const str = this.full_product_name;
+        const regex = /\(([^()]+)\)[^(]*$/;
+        const matches = str.match(regex);
 
+        if (matches && matches.length > 1) {
+            const attributes = matches[matches.length - 1].trim();
+            const productName = str.replace(matches[0], "").trim();
+            return { productName, attributes };
+        }
+
+        return { productName: str, attributes: "" };
+    }
```

Comment:
Not needed anymore look at `attribute_value_ids` and use attribute formater is needed `attributeFormatter`